conda_smithy/lint_recipe.py

```diff
@@ -149,9 +155,55 @@ def lintify(meta, recipe_dir=None):
     except RuntimeError as e:
         lints.append(str(e))
 
+    # 13: Check that the recipe_dir has the same name as the recipe_name
+    recipe_name = package_section.get('name', '').strip()
+    if recipe_dir:
```

Comment:
So I don't think we want to enforce this in the feedstocks. As for staged-recipes, it's suppose to not care what the directories are named any more. ( https://github.com/conda-forge/staged-recipes/pull/2313 ) Have there been cases where this is not working?